pkg/api/convertertoapi.go

```diff
@@ -718,12 +718,15 @@ func setVlabsKubernetesDefaults(vp *vlabs.Properties, api *OrchestratorProfile)
 
 	if vp.OrchestratorProfile.KubernetesConfig != nil {
 		// Included here for backwards compatibility with deprecated NetworkPolicy usage patterns
-		if vp.OrchestratorProfile.KubernetesConfig.NetworkPolicy == NetworkPolicyAzure {
-			api.KubernetesConfig.NetworkPlugin = vp.OrchestratorProfile.KubernetesConfig.NetworkPolicy
-			api.KubernetesConfig.NetworkPolicy = "" // no-op but included for emphasis
-		} else if vp.OrchestratorProfile.KubernetesConfig.NetworkPolicy == NetworkPolicyNone {
+		if vp.OrchestratorProfile.KubernetesConfig.NetworkPolicy == NetworkPolicyNone {
 			api.KubernetesConfig.NetworkPlugin = NetworkPluginKubenet
 			api.KubernetesConfig.NetworkPolicy = "" // no-op but included for emphasis
+		} else if vp.OrchestratorProfile.KubernetesConfig.NetworkPolicy == NetworkPolicyAzure {
```

Comment:
I think this should happen in defaults.go since it is modifying the default behavior